### ISSUE TITLE
"COVID-19 Exposure Logging - Weekly Update" - Strange information

### ISSUE THREAD

**Issue by irieger** [corona-warn-app/cwa-app-ios#1096]
I understand this isn't directly (only) a thing of CWA but as there is no real way to contact Apple and I would hope you are in contact with Apple about the framework and ways to improve it and the CWA UX. (The information in the FAQ is a bit sparse I think)

## Describe the bug

The Apple ENF sends those weekly updates as you are well aware. Until today it was always "... identified 0 potential exposures ..." which I think is already misleading as any contact to another person is a possible exposure unless we can be sure those contacts are 100% covid free.
Today this changed: The counter suddenly changed to 5 so I had "5 potential exposures". When clicking the notification one lands inside the iOS COVID-19 settings dialog instead of the app which to me would be the logic target. But even worse you don't see any information in this settings dialog. The app state is green with "no exposure up to now".
Clicking through some of the checks in the log I saw only something like "Matched Key Count 1" on the topmost entry on 24.08., the same on the second entry on the 25. etc. Aren't this all the same exposures that then seem to be rechecked every day? This one match was counted 5 times until it wasn't matched this night so I know which day this matched might have occured.

## Expected behaviour

Give more verbose information in the app. Like: The ENF matched 5 keys but you haven't been in close proximity...

That simplified interface is more annoying than verbosity.

## Steps to reproduce the issue

Be near enough to someone who is later identified as expose but be outside the RKI parameters...

## Technical details

- iOS Version: 13.6.1
- Device: iPhone 7 Plus

## Possible Fix

More clarity and verbosity in the user interface and more open information about what happens behind the curtain.

## Additional context

I think this kind of UX is more annoying and terrifying than not having the app although of course this for me is no reason to throw away the app.
Haven't been out in a sense of meeting people I don't know since Corona started but was on a train ride two weeks ago so maybe that match was there which with the current press about unreliability in trains ...

## PS

While I'm on it you should in general talk better with Apple. They should for example auto-start each Covid-App at first download to trigger the dialog wheter to activate. And allow more regular tests than once a day and in case of a match give more information to be able to get a rough idea of what one did at that day to better understand the detected situation. Maybe the "v2" of the API coming with iOS 14/13.7 improves on this front somehow.

----
Internal Tracking ID: EXPOSUREAPP-2532

**Comment by daimpi** [comment]
Regarding the iOS weekly update: 
>The plan is to rework this message as soon as possible.

([link](https://github.com/corona-warn-app/cwa-app-ios/issues/1011#issuecomment-675310587))

In general you might want to check out this issue, as there seems to be currently a lot of confusion surrounding those "weekly updates" by iOS: https://github.com/corona-warn-app/cwa-app-ios/issues/1011
Just out of curiosity: did you check your CWA on one of the days where the ENF log shows a match and if so: was it showing you "1 Exposure with low risk (green)" on this day?

If you want, you can also share your ENF log here or over at https://github.com/corona-warn-app/cwa-app-ios/issues/1011, then we could look at it and try to determine how the "weekly update" by iOS might have come about.
Also: if you want a quick look at those encounters in your ENF logo, this tool is pretty neat: https://felixlen.github.io/ena_log/ 🙂. 

Regarding an improvement in the sense that more information is shown on green encounters: I fully agree and I hope we see some movement by RKI in this respect. Relevant threads on this topic:
- https://github.com/corona-warn-app/cwa-backlog/issues/23
- https://github.com/corona-warn-app/cwa-wishlist/issues/100

**Comment by irieger** [comment]
I think with what I found out and now saw in the linked threads that my search for weekly update and other keywords didn't find it is expected behaviour for what it is now with the ENF and my info doesn't add much new to the other thread so I append it here.

From the linked ena_log tool I got what I already found out:
Anzahl Treffer | Überprüfungszeitpunkt
-- | --
1 | 8/24/2020, 5:16 PM
1 | 8/25/2020, 5:25 PM
1 | 8/26/2020, 5:54 PM
1 | 8/27/2020, 6:16 PM
1 | 8/28/2020, 7:48 PM
0 | 8/30/2020, 12:23 AM
0 | 8/30/2020, 12:23 AM

Key showed up on 24th and is now out of scope.
Unfortunately I didn't look in the app for some days. I keep an eye on the exposure checks log every few days but since 13.6.1 and the most recent CWA update I haven't had a single hickup (after manually starting it once after the update, which shouldn't be required). Was expecting to get notifications in case of exposures.

**Comment by daimpi** [comment]
Yes, your log seems to perfectly fit the "iOS weekly update adds up all the matches of the last 7 days" explanation.

Given that the match disappeared today, the encounter probably happened on 2020-08-15.  

>Was expecting to get notifications in case of exposures.

You only get a push notification from CWA in case of a "increased risk (red) exposure".

**Comment by irieger** [comment]
> Given that the match disappeared today, the encounter probably happened on 2020-08-15.

That's what I figured. So it was nearly guaranteed to have been an outdoor passing by on this day based on my activity. Therefore it would be nice to have at least the day. One day of hiking for me is completely different to one day of 3 hours train rides ...
 
> You only get a push notification from CWA in case of a "increased risk (red) exposure".

Would be nice to give the user some self responsibility ("Mündigkeit") and don't hide stuff. The UI shows it anyway so why not be open?

**Comment by daimpi** [comment]
> Would be nice to give the user some self responsibility ("Mündigkeit") and don't hide stuff. The UI shows it anyway so why not be open?

I agree… I think the user should at least have the option to get more information displayed. I've suggested exactly this e.g. [here](https://github.com/corona-warn-app/cwa-backlog/issues/23#issuecomment-678768765) (ignore my statement that "CWA hides encounters", there was confusion around this topic, but it has since been tested and found out that all encounters are shown in CWA 😅).

**Comment by irieger** [comment]
> I agree… I think the user should at least have the option to get more information displayed. I've suggested exactly this e.g. [here](https://github.com/corona-warn-app/cwa-backlog/issues/23#issuecomment-678768765) (ignore my statement that "CWA hides encounters", there was confusion around this topic, but it has since been tested and found out that all encounters are shown in CWA sweat_smile).

Good idea, an expert mode is exactly what came to my mind yesterday when seeing how the app is trimmed to keep it simple stupid.

**Comment by Marco2907** [comment]
Hello @irieger , thank you for sharing your content. I will go forward with our development team to discuss this point based on a duplicate regarding to #1011

Thank you.

Best,
MP

Corona-Warn-App Open Source Team

**Comment by Marco2907** [comment]
Hi @irieger, thanks for sharing your content and your proposal for the additional information inside the CWA app. I will keep this point to discuss this with our development team. I will come back to you as soon as possible. 

Thanks,
MP

Corona-Warn-App Open Source Team

---
Internal Tracking ID: EXPOSUREAPP-2532

**Comment by irieger** [comment]
I add one more question here as today I realized it is not clear to me and is related to this message: What does one encounter mean?
Until today I was under the assumption that one encounter means one 15 Minute block matching. Then I heard Linus Neumann today in the https://logbuch-netzpolitik.de/ Podcast and he talked about his low-risk encounters number. And they seemed to assume it was only counting the number of daily keys that matched one or more 15 minute keys? If this is so I would also expect the length of the encounter, which otherwise I could roughly assume from the number of encounters.

**Comment by Ein-Tim** [comment]
@irieger Did you see [this ](https://www.coronawarn.app/en/faq/#encounter_but_green) FAQ entry?

**Comment by daimpi** [comment]
> I add one more question here as today I realized it is not clear to me and is related to this message: What does one encounter mean?

Encounter  just means that you recorded a rolling proximity identifier (RPI) of someone who later tested positive and uploaded their keys.

For an encounter per se it is [irrelevant how much time you spent/at which distance you were](https://github.com/corona-warn-app/cwa-website/issues/307#issuecomment-684774771) wrt the infected person. Those factors only become relevant when it comes to [determining whether you had a "red-/increased-risk" encounter](https://github.com/corona-warn-app/cwa-documentation/blob/master/cwa-risk-assessment.md).

Unfortunately there is currently a lot of misinformation surrounding this issue, especially since the [FAQ entry](https://www.coronawarn.app/en/faq/#encounter_but_green) Tim mentions above got an update which made it straight up false, see my **Edit4** [here](https://github.com/corona-warn-app/cwa-website/issues/307).

@Ein-Tim I would try to avoid guiding ppl to this FAQ entry w/o further explanation, until it is fixed. Otherwise it's just further spreading the misinformation.

**Comment by Ein-Tim** [comment]
@daimpi You are so right! I totally forgot about this wrong entry. Please could somebody from the devs (@mtb77 @svengabr) come back to us [here](asap). Thanks! And thanks for the reminder @daimpi 😀👍

**Comment by irieger** [comment]
Saw that not so helpful FAQ entry. It doesn't answer anything.

> For an encounter per se it is [irrelevant how much time you spent/at which distance you were](https://github.com/corona-warn-app/cwa-website/issues/307#issuecomment-684774771) wrt the infected person. Those factors only become relevant when it comes to [determining whether you had a "red-/increased-risk" encounter](https://github.com/corona-warn-app/cwa-documentation/blob/master/cwa-risk-assessment.md).

But it might be interesting for me knowing if it was one short key recording of one RPI or if I saw multiple RPI from the same person. When I was outdoor and just passed a person on the street exactly at the RPI recording interval so I only saw the key once is something different than having the same person around for like 2 hours. I think the user knows best in which situations she/he was at that day and should at least get the day and length of this encounter.

So is the encounter one person (cumulative over the RPI matched to the days key) or each individual RPI? That is still not clear from your answers.

**Comment by svengabr** [comment]
@Ein-Tim @daimpi I saw your feedback regarding the FAQ entries and Risk Encounters. I can assure this is not forgotten. There are currently 5 open issues regarding the same topic and there will be feedback soon. The RKI is working out a new text which isn't final yet. The text changes are expected to come to FAQ and CWA with the next hotfix version after 1.3.0.

**Comment by daimpi** [comment]
@irieger 
> But it might be interesting for me knowing if it was one short key recording of one RPI or if I saw multiple RPI from the same person. When I was outdoor and just passed a person on the street exactly at the RPI recording interval so I only saw the key once is something different than having the same person around for like 2 hours. I think the user knows best in which situations she/he was at that day and should at least get the day and length of this encounter.

Yes I agree, that's why we have issues like 
- https://github.com/corona-warn-app/cwa-backlog/issues/23
- https://github.com/corona-warn-app/cwa-wishlist/issues/100
- https://github.com/corona-warn-app/cwa-wishlist/issues/178

where all of this can be discussed in more detail 🙂. 

> So is the encounter one person (cumulative over the RPI matched to the days key) or each individual RPI? That is still not clear from your answers.

I think I now better understand what your question was. 
tl;dr: **one** encounter means one or more matches with **one** 24h valid key (DK) that someone who tested positive shared.

Long version: when you download the Diagnosis Keys (DKs)/Temporary Exposure Keys(TEKs)  from the server those have a validity window of 24h. Each positive person shares up to 14 DKs for the last 14 days before their diagnosis/upload. Your phone expands each TEK/DK (I'm using the terms DK & TEK interchangeably) it downloaded to 144 RPIs which are then compared to the RPIs your phone saw/recorded ([via](https://covidapps.slack.com/archives/C0194ML0MLN/p1598700303243800?thread_ts=1598688186.242400&cid=C0194ML0MLN) @mh-, Slack [invitation link](https://join.slack.com/t/covidapps/shared_invite/zt-gmgl4l13-E4Xcn1ZfPFsPz6V_KjROCg)):
> It’s explained here: https://blog.google/documents/69/Exposure_Notification_-_Cryptography_Specification_v1.2.1.pdf
You are right that from one TEK, 144 RPIs are derived. 144x10min == 1440 min == 24x60min.
Each RPI is based on the TEK as well as the ENIN which represents this 10 minute interval.
So we know when this RPI should have been generated. It is only accepted as a match, if it was received in a +/- 2 hours time window around the expected generation time.

You can also see this process described [here ](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md)(in particular in Figure 11) in the CWA documentation.

So one match in your ENF log/one encounter in your CWA can mean anything from 1 - 144 RPIs matched, as long as they all belong to the same DK/TEK. When you matched with RPIs (no matter the number) from two different TEKs you'd get 2 matches in your ENF log/ 2 encounters in CWA and so forth…

**Comment by Ein-Tim** [comment]
@svengabr Why is this a Hot Topic Issue? I would rather say it is solved 😅

**Comment by heinezen** [comment]
@Ein-Tim I think this was labelled by accident. Removed the label and closed the issue, since it was addressed by Apple and has an [FAQ entry](https://www.coronawarn.app/en/faq/#notif_weekly_update) with info for turning it off completely.
